src/DonorDashboards/Tabs/DonationHistoryTab/DonationsRoute.php

```diff
@@ -52,6 +52,30 @@ public function handleRequest( WP_REST_Request $request ) {
 	 * @return WP_REST_Response
 	 */
 	protected function getData( DonationsRepository $repository, $donorId ) {
+
+		// Check that the provided donor ID is valid
+		if ( Give()->donors->get_donor_by( 'id', $donorId ) !== false ) {
+			Log::error(
+				esc_html__( 'An error occurred while retrieving donation records', 'give' ),
+				[
+					'source'   => 'Donor Dashboard',
+					'Donor ID' => $donorId,
+					'Error'    => __( 'An invalid Donor ID was provided.', 'give' ),
+				]
+			);
+
+			return new WP_REST_Response(
+				[
+					'status'        => 400,
+					'response'      => 'database_error',
+					'body_response' => [
+						'message' => esc_html__( 'An error occurred while retrieving your donation records.', 'give' ),
```

Comment:
@henryholtgeerts For end-user facing errors like this, please give the end-user an actionable next step. Even if it's to reach out to the website owner and let them know.

Reply:
Good call! Done, ref: fix: add actionable item to message